### ISSUE TITLE
how to use eval_g1_dataset.py

### ISSUE THREAD

**Issue by bangboomtata** [unitreerobotics/unitree_IL_lerobot#63]
I ran the eval_g1_dataset.py as below, and it just showed a progress bar until 100% and finish evaluating. I am wondering what is the use of this script?

python unitree_lerobot/eval_robot/eval_g1_dataset.py  \
    --policy.path=unitree_lerobot/lerobot/outputs/train/2025-12-30/13-17-18_act/checkpoints/100000/pretrained_model \
    --repo_id=bangboomtata/robotics-demo \
    --root="" \
    --episodes=0 \
    --frequency=30 \
    --arm="G1_29" \
    --ee="inspire1" \
    --visualization=true \
    --send_real_robot=false
INFO 2025-12-31 11:23:14 _dataset.py:176 {'arm': 'G1_29',
 'ee': 'inspire1',
 'episodes': 0,
 'frequency': 30.0,
 'headless': False,
 'motion': False,
 'policy': {'chunk_size': 100,
            'device': 'cuda',
            'dim_feedforward': 3200,
            'dim_model': 512,
            'dropout': 0.1,
            'feedforward_activation': 'relu',
            'input_features': {'observation.images.cam_high': {'shape': (3,
                                                                         480,
                                                                         640),
                                                               'type': <FeatureType.VISUAL: 'VISUAL'>},
                               'observation.images.cam_left_wrist': {'shape': (3,
                                                                               480,
                                                                               640),
                                                                     'type': <FeatureType.VISUAL: 'VISUAL'>},
                               'observation.images.cam_right_wrist': {'shape': (3,
                                                                                480,
                                                                                640),
                                                                      'type': <FeatureType.VISUAL: 'VISUAL'>},
                               'observation.state': {'shape': (26,),
                                                     'type': <FeatureType.STATE: 'STATE'>}},
            'kl_weight': 10.0,
            'latent_dim': 32,
            'license': None,
            'n_action_steps': 100,
            'n_decoder_layers': 1,
            'n_encoder_layers': 4,
            'n_heads': 8,
            'n_obs_steps': 1,
            'n_vae_encoder_layers': 4,
            'normalization_mapping': {'ACTION': <NormalizationMode.MEAN_STD: 'MEAN_STD'>,
                                      'STATE': <NormalizationMode.MEAN_STD: 'MEAN_STD'>,
                                      'VISUAL': <NormalizationMode.MEAN_STD: 'MEAN_STD'>},
            'optimizer_lr': 1e-05,
            'optimizer_lr_backbone': 1e-05,
            'optimizer_weight_decay': 0.0001,
            'output_features': {'action': {'shape': (26,),
                                           'type': <FeatureType.ACTION: 'ACTION'>}},
            'pre_norm': False,
            'pretrained_backbone_weights': 'ResNet18_Weights.IMAGENET1K_V1',
            'pretrained_path': 'unitree_lerobot/lerobot/outputs/train/2025-12-30/13-17-18_act/checkpoints/100000/pretrained_model',
            'private': None,
            'push_to_hub': False,
            'replace_final_stride_with_dilation': 0,
            'repo_id': None,
            'tags': None,
            'temporal_ensemble_coeff': None,
            'use_amp': False,
            'use_vae': True,
            'vision_backbone': 'resnet18'},
 'rename_map': {},
 'repo_id': 'bangboomtata/robotics-demo',
 'root': '',
 'send_real_robot': False,
 'use_dataset': False,
 'visualization': True}
INFO 2025-12-31 11:23:14 _dataset.py:184 Making policy.
Loading weights from local directory
11:23:15:519720 INFO     Arguments: EvalRealConfig(repo_id='bangboomtata/robotics-demo', policy=ACTConfig(n_obs_steps=1, input_features={'observation.state':       eval_g1_dataset.py:58
                         PolicyFeature(type=<FeatureType.STATE: 'STATE'>, shape=(26,)), 'observation.images.cam_high': PolicyFeature(type=<FeatureType.VISUAL:                           
                         'VISUAL'>, shape=(3, 480, 640)), 'observation.images.cam_left_wrist': PolicyFeature(type=<FeatureType.VISUAL: 'VISUAL'>, shape=(3, 480,                         
                         640)), 'observation.images.cam_right_wrist': PolicyFeature(type=<FeatureType.VISUAL: 'VISUAL'>, shape=(3, 480, 640))},                                          
                         output_features={'action': PolicyFeature(type=<FeatureType.ACTION: 'ACTION'>, shape=(26,))}, device='cuda', use_amp=False,                                      
                         push_to_hub=False, repo_id=None, private=None, tags=None, license=None,                                                                                         
                         pretrained_path='unitree_lerobot/lerobot/outputs/train/2025-12-30/13-17-18_act/checkpoints/100000/pretrained_model', chunk_size=100,                            
                         n_action_steps=100, normalization_mapping={'VISUAL': <NormalizationMode.MEAN_STD: 'MEAN_STD'>, 'STATE': <NormalizationMode.MEAN_STD:                            
                         'MEAN_STD'>, 'ACTION': <NormalizationMode.MEAN_STD: 'MEAN_STD'>}, vision_backbone='resnet18',                                                                   
                         pretrained_backbone_weights='ResNet18_Weights.IMAGENET1K_V1', replace_final_stride_with_dilation=0, pre_norm=False, dim_model=512,                              
                         n_heads=8, dim_feedforward=3200, feedforward_activation='relu', n_encoder_layers=4, n_decoder_layers=1, use_vae=True, latent_dim=32,                            
                         n_vae_encoder_layers=4, temporal_ensemble_coeff=None, dropout=0.1, kl_weight=10.0, optimizer_lr=1e-05, optimizer_weight_decay=0.0001,                           
                         optimizer_lr_backbone=1e-05), root='', episodes=0, frequency=30.0, arm='G1_29', ee='inspire1', motion=False, headless=False,                                    
                         visualization=True, send_real_robot=False, use_dataset=False, rename_map={})                                                                                    
Please enter the start signal (enter 's' to start the subsequent program):s
  0%|                                                                                                                                                             | 0/49 [00:00<?, ?it/s]RerunLogger: First data packet received. Auto-configuring...
  - Using 'index' for time sequence.
  - Detected State Key: 'observation.state'
  - Detected Action Key: 'action'
  - Detected Image Keys: ('observation.images.cam_high', 'observation.images.cam_left_wrist', 'observation.images.cam_right_wrist')
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 49/49 [00:05<00:00,  8.80it/s]
INFO 2025-12-31 11:23:25 _dataset.py:204 End of eval
(unitree_lerobot) user@PG15HL01BB:~/workspace/unitree_IL_lerobot$

**Comment by hengguo-boop** [comment]
Sorry for replying to this question so late. 
The function of this script is to conduct open-loop testing on the dataset, loading the status and images of the data, and comparing the predicted actions of the model with the real actions in the dataset. 
The main purpose is to check the quality of the model training and prevent any potential damage to the real machine when it is directly used. Of course, I must clarify that excellent performance on the dataset does not necessarily mean the same effect on the real machine.
